File: trab3/python/iteracaolinear.py
from base import *


@algoritmo
def iteracaolinear(self, f, x0):
    g = lambda x: f(x) + x
    x1 = g(x0)
    self.tabela('k', 'x0', 'x1', 'g(x1)')
    for k in xrange(self.nmi):
        x2 = g(x1)
        self.tabela(k, x0, x1, x2)
        if abs(x2 - x1) < self.tol and abs(x1 - x0) < self.tol:
            return x1
        x0, x1 = x1, x2
    raise ErroNMI

File: trab3/ruby/iteracaolinear.rb
# encoding: utf-8
# Algoritmo ITERAÇÃO LINEAR
require "./base"

Calcnum::algoritmo :iteracaolinear do |x0|
  def g(x) f(x) + x; end
  x1 = g(x0)
  tabela 'k', 'x0', 'x1', 'g(x1)'
  iterar do |k|
    x2 = g(x1)
    tabela k, x0, x1, x2
    break x1 if (x2 - x1).abs < tol and (x1 - x0).abs < tol
    x0, x1 = x1, x2
  end
end

